pyglet/experimental/multitexture_sprite.py

```diff
@@ -0,0 +1,331 @@
+
+    The Group defines custom ``__eq__`` and ``__hash__`` methods, and so will be
+    automatically coalesced with other MultiTexture Sprite Groups sharing the same
+    parent Group, Textures, and blend parameters.
+    """
+
+    def __init__(self, textures: dict[str, Texture], blend_src: int, blend_dest: int,
+                 program: ShaderProgram | None = None, parent: Group | None = None) -> None:
+        """Create a sprite group for multiple textures and samplers.
+
+        All textures must share the same target type.  The group is created internally
+        when a :py:class:`~pyglet.multitexture_sprite.MultiTextureSprite` is created;
+        applications usually do not need to explicitly create it.
+
+        Args:
+            textures:
+                A dictionary of textures, with the keys being the GLSL sampler name.
+            blend_src:
+                OpenGL blend source mode; for example, ``GL_SRC_ALPHA``.
+            blend_dest:
+                OpenGL blend destination mode; for example, ``GL_ONE_MINUS_SRC_ALPHA``.
+            program:
+                A custom ShaderProgram.
+            parent:
+                Optional parent group.
+        """
+        self.textures = textures
+        texture = list(self.textures.values())[0]
+        super().__init__(texture, blend_src, blend_dest, program, parent)
+
+    def set_state(self) -> None:
+        self.program.use()
+
+        for idx, name in enumerate(self.textures):
+            self.program[name] = idx
+
+        for i, texture in enumerate(self.textures.values()):
+            glActiveTexture(GL_TEXTURE0 + i)
+            glBindTexture(texture.target, texture.id)
+
+        glEnable(GL_BLEND)
+        glBlendFunc(self.blend_src, self.blend_dest)
+
+    def unset_state(self) -> None:
+        glDisable(GL_BLEND)
+        self.program.stop()
+        glActiveTexture(GL_TEXTURE0)
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}({self.texture}-{self.texture.id})'
+
+    def __eq__(self, other: object | Group | MultiTextureSpriteGroup) -> bool:
+        """Determines if this group is the same as the other group.
+
+        The method is broken out like this to make it easier to follow and read.
+        """
+        if other.__class__ is not self.__class__:
+            return False
+
+        # Check targets and ids
+        for name, texture in self.textures.items():
+            if name not in other.textures:
+                return False
+
+            if texture.id != other.textures[name].id or texture.target != other.textures[name].target:
+                return False
+
+        # Made it this far so just check the remainder
+        return (self.program is other.program and
+                self.blend_src == other.blend_src and
+                self.blend_dest == other.blend_dest)
+
+    def __hash__(self) -> int:
+        tex_id = tuple([texture.id for texture in self.textures.values()])
```

Comment:
Maybe a performance speedup to cache this on group initialization? That way it's not making tuples every time it's hash checked?